tests/common/utils/test_distro_version.py

```diff
@@ -0,0 +1,128 @@
+
+from azurelinuxagent.common.utils.distro_version import DistroVersion
+from azurelinuxagent.common.utils.flexible_version import FlexibleVersion
+
+
+class TestDistroVersion(AgentTestCase):
+
+    def test_it_should_implement_all_comparison_operators(self):
+        self.assertTrue(DistroVersion("1.0.0") < DistroVersion("1.1.0"))
+        self.assertTrue(DistroVersion("1.0.0") <= DistroVersion("1.0.0"))
+        self.assertTrue(DistroVersion("1.0.0") <= DistroVersion("1.1.0"))
+
+        self.assertTrue(DistroVersion("1.1.0") > DistroVersion("1.0.0"))
+        self.assertTrue(DistroVersion("1.1.0") >= DistroVersion("1.1.0"))
+        self.assertTrue(DistroVersion("1.1.0") >= DistroVersion("1.0.0"))
+
+        self.assertTrue(DistroVersion("1.1.0") != DistroVersion("1.0.0"))
+        self.assertTrue(DistroVersion("1.1.0") == DistroVersion("1.1.0"))
+
+    def test_it_should_compare_digit_sequences_numerically(self):
+        self.assertTrue(DistroVersion("2.0.0") < DistroVersion("10.0.0"))
+        self.assertTrue(DistroVersion("1.2.0") < DistroVersion("1.10.0"))
+        self.assertTrue(DistroVersion("1.0.2") < DistroVersion("1.0.10"))
+        self.assertTrue(DistroVersion("2.0.rc.2") < DistroVersion("2.0.rc.10"))
+        self.assertTrue(DistroVersion("2.0.rc2") < DistroVersion("2.0.rc10"))
+
+    def test_it_should_compare_non_digit_sequences_lexicographically(self):
+        self.assertTrue(DistroVersion("2.0.alpha") < DistroVersion("2.0.beta"))
+        self.assertTrue(DistroVersion("2.0.alpha.2") < DistroVersion("2.0.beta.1"))
+        self.assertTrue(DistroVersion("alpha") < DistroVersion("beta"))
+        self.assertTrue(DistroVersion("<1.0.0>") < DistroVersion(">1.0.0>"))
+
+    def test_it_should_parse_common_distro_versions(self):
+        """
+        Test that DistroVersion can parse the versions given by azurelinuxagent.common.version.DISTRO_VERSION
+        (the values in distro_versions.txt are current values from telemetry.)
+        """
+        data_file = os.path.join(data_dir, "distro_versions.txt")
+
+        with open(data_file, "r") as f:
+            for line in f:
+                line = line.rstrip()
+                version = DistroVersion(line)
+                self.assertNotEqual([], version._fragments)
+
+        self.assertEqual([], DistroVersion("")._fragments)
+
+    def test_it_should_compare_commonly_used_versions(self):
+        """
+        Test that DistroVersion does some common comparisons correctly.
+        """
+        self.assertTrue(DistroVersion("1.0.0") < DistroVersion("2.0.0."))
+        self.assertTrue(DistroVersion("1.0.0") < DistroVersion("1.1.0"))
+        self.assertTrue(DistroVersion("1.0.0") < DistroVersion("1.0.1"))
+
+        self.assertTrue(DistroVersion("1.0.0") == DistroVersion("1.0.0"))
+        self.assertTrue(DistroVersion("1.0.0") != DistroVersion("2.0.0"))
+
+        self.assertTrue(DistroVersion("13") != DistroVersion("13.0"))
```

Comment:
This looks to me DistroVersion is also variant of strict version and Flexible version also documented as same. Is that means both does same thing? 

13 != 13.0  !=13.0.0 is true in both modules?

Reply:
DistroVersion is not a variant of StrictVersion, but of LooseVersion